packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -756,24 +792,21 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         channel.handle.attachGraph();
```

Comment:
Any chance you can add asserts here clearly articulating the state of this data store? I think it has to be globally visible / attached (latter is asserted, but former is not).
May that assert should be in isAttached() itself, when it returns true (and maybe more asserts to clearly spell out valid invariants)?

Other related question - can we have only one enum to track all the states? I.e. get rid of this._attachState, and everything else be a projection from that enum? To avoid even thinking about cases (if it's possible to get to them) like attached but not locally visible.
public isAttached(): boolean

Reply:
The idea is that we can remove bind, graph attach and attach state and only use visibility state. We may still have to expose attachState because its on the public interfaces but we can derive that from the visibility state.
We can also remove attachState and replace it with visibility state but I think that is a bigger breaking change.

And yes, we can mostly assert that the state is globally visible here. Let me work on that.